presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerde.java

```diff
@@ -88,6 +89,41 @@ public SerializedPage serialize(Page page)
         return new SerializedPage(slice, markers, page.getPositionCount(), uncompressedSize);
     }
 
+    public SerializedPage serialize(Slice slice, int positionCount)
+    {
+        checkArgument(slice.isCompact(), "slice is not compact");
+
+        int uncompressedSize = slice.length();
```

Comment:
They are not exactly the same. The new serialize(Slice slice, int positionCount) has additional optimizations that use byte[] instead of ByteBuffer as the compressionBuffer and make it a class member to avoid allocating new memory each time. So to compare the performance of the two operators I'd like to make them separate.

Reply:
@yingsu00 Thanks for explaining. Looks like this is a small change that existing operator can benefit from. Any reason not to apply this optimization and reap some benefits earlier? You can always put old code into a benchmark to test/confirm the gains.

Reply:
Sure I can cut a separate PR for it if that's what you're talking about.

Reply:
@yingsu00 That would be great!

Reply:
https://github.com/prestodb/presto/pull/13232

Reply:
@wenleix 

> Also note this method, in theory, is not doing "serialize": the page is already serialized into slice. Would wrapSerializedSlice a better name? Do you have any suggestion? @mbasmanova

I agree that the name could be improved. I don't have a good suggestion though. For now, I think it would be reasonable to use the same name as an existing method and unify their implementations. A rename of both methods can happen separately.

Reply:
Added a new commit c8eb31102d "Extract wrapSlice method in PagesSerde" as @wenleix suggested.